parlai/core/torch_agent.py

```diff
@@ -250,6 +250,45 @@ def vectorize(self, obs, add_start=True, add_end=True, truncate=None):
 
         return obs
 
+    def _padded_tensor(self, items):
+        """Create a right-padded matrix from an uneven list of lists.
+
+        :param list[list[int]] items: List of items
+        :param bool sort: If True, orders by the length
+        :rtype: (Tensor[int64], list[int])
+        :return: (padded, lengths)
+        """
+        n = len(items)
+        lens = [len(item) for item in items]
+        t = max(lens)
+        output = torch.LongTensor(n, t).fill_(self.NULL_IDX)
+        for i in range(len(items)):
+            output[i, :lens[i]] = items[i]
+        if self.use_cuda:
+            output = output.cuda()
+        return output, lens
+
+    def _argsort(self, keys, *lists, descending=False):
```

Comment:
TODO: unit test.

Both these utility methods could be useful as general utilities rather than being stuffed into this module here.